src/rosdep2/main.py

```diff
@@ -734,7 +734,7 @@ def command_check(lookup, packages, options):
                 print('ERROR[%s]: resource not found [%s]' % (package_name, ex.args[0]), file=sys.stderr)
             else:
                 print('ERROR[%s]: %s' % (package_name, ex), file=sys.stderr)
-    if uninstalled:
+    if uninstalled or errors:
```

Comment:
```suggestion
    if errors:
        return 2
    if uninstalled:
```

The or means that the return 1 won't be hit. I initially was going to suggest an and, but we still want an error code if there are errors but no uninstalled. So I think just having one code for errors, and a second for no errors but uninstalled packages.

Reply:
~~I thought about this too, but pretty much every other return code indicating failure was `1`.~~
~~I was hesitant to add another number - what do you say?~~

After some digging, it seems that the behavior is not consistent.
- In [init](https://github.com/ros-infrastructure/rosdep/blob/e48589c676e45f7ef90599edd86b742de3a9c9ef/src/rosdep2/main.py#L601), each error has its code.
- In [update](https://github.com/ros-infrastructure/rosdep/blob/e48589c676e45f7ef90599edd86b742de3a9c9ef/src/rosdep2/main.py#L637), they're all 1.

Since `check` only had 1 type of error, I think I'll change to `return 2`.

Reply:
The conditional block immediately preceding the `if uninstalled:` line was already an `if errors:` block which is why I suggested the change in that way, but this change works as well.